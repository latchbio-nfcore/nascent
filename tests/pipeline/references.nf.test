def igenomes_base = 's3://ngi-igenomes/igenomes'

nextflow_pipeline {

    name "Test passing references"
    script "main.nf"

    test("Should work with BWA Index") {
        // https://github.com/nf-core/nascent/issues/106
        when {
            params {
                outdir = "$outputDir"
                aligner = "bwa"
                bwa_index = "${igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/"
                fasta = "${igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/WholeGenomeFasta/genome.fa"
                gtf = "${igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf"
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                workflow.trace.tasks().size(),
                path("$outputDir/bbmap").list(),
                path("$outputDir/bed2saf").list(),
                // BUG https://github.com/nf-core/nascent/issues/102
                // path("$outputDir/bedtools").list(),
                // FIXME gtf2bed isn't deterministic
                // path("$outputDir/bwa/featurecounts/gene").list(),
                // FIXME cd4 isn't deterministic
                // path("$outputDir/bwa/featurecounts/predicted").list(),
                // path("$outputDir/deeptools").list(),
                // path("$outputDir/fastp/*.json").list(),
                // FIXME gtf2bed isn't deterministic
                // path("$outputDir/gtf2bed").list(),
                path("$outputDir/homer/cd4.bed"),
                path("$outputDir/homer/jurkat.bed"),
                path("$outputDir/preseq").list(),
                // path("$outputDir/rseqc").list(),
                path("$outputDir/samtools").list(),
            ).match()
        }

    }

}
