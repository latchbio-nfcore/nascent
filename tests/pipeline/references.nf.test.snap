{
    "Should work with gzipped references": {
        "content": [
            99
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-14T16:39:27.539091004"
    },
    "Should work with BWA Index": {
        "content": [
            126
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-15T09:38:13.33349927"
    }
}